### ISSUE TITLE
About layer normalization

### ISSUE THREAD

**Issue by WilliamTambellini** [intel/mkl-dnn#511]
Hello,
I have searched on
https://intel.github.io/mkl-dnn/
and also in mkldnn sources but could nt find any : 
is there somewhere a mkldnn implementation of layer normalization as defined in : 
https://arxiv.org/pdf/1607.06450
https://arxiv.org/abs/1607.06450
Note : "batch norm" is different from "layer norm" even if both are close :
https://nealjean.com/ml/neural-network-normalization/

The TF API:
https://www.tensorflow.org/api_docs/python/tf/contrib/layers/layer_norm
A TF imp: 
https://github.com/tensorflow/tensorflow/pull/6205/files

**Comment by irinasok** [comment]
Hi,

That's right, Layer Normalization is not implemented in MKL-DNN yet. We are working on it at the moment.

Please note that there are different flavors of Layer Norm implemented in different frameworks. Our plan is to implement similar to Batch Norm API but it will perform normalization over the **last logical axis**.  That contradicts to TF implementation that allows LN to normalize over several dimensions (begin_norm_axis ... last_axis). We don't expect it to be a big limitation due to the fact that LN is used in NLP models where tensors have only one feature dimension.

**Comment by WilliamTambellini** [comment]
ok thanks @irinasok

**Comment by vpirogov** [comment]
Layer normalization is added in f462235fbb90491e44a36967f08834a1f97453a1. (Corresponding section)[http://intel.github.io/mkl-dnn/dev_guide_layer_normalization.html] is added to developer guide as well.

**Comment by WilliamTambellini** [comment]
Thank you.
